data.py

```diff
@@ -1,92 +1,116 @@
 
         Data comes into the class in the V2/V3 MQTT formats and may be accessed
         by using this class as a dictionary. This class is implemented by
         versions for specific bikes (DataV2, DataV3,...) """
 
-    data_types = {
-        # DAS data
-        "power": int,
-        "cadence": int,
-        "heartRate": int,
-        "gps": int,
-        "gps_speed": float,
-        "reed_velocity": float,
-        "reed_distance": float,
-        # Power model data
-        "rec_power": float,
-        "rec_speed": float,
-        "predicted_max_speed": float,
-        "zdist": float,
-        "plan_name": str,
-    }
-
     def __init__(self):
         # This is by no means a complete list of data fields we could track -
         # just the ones we currently think we might use on the overlays.
         self.data = {
             # DAS data
-            "power": 0,
-            "cadence": 0,
-            "heartRate": 0,
-            "gps": 0,
-            "gps_speed": 0,
-            "reed_velocity": 0,
-            "reed_distance": 0,
+            "power": DataValue(int),
+            "cadence": DataValue(int),
+            "heartRate": DataValue(int),
+            "gps": DataValue(int),
+            "gps_speed": DataValue(float),
+            "reed_velocity": DataValue(float),
+            "reed_distance": DataValue(float),
             # Power model data
-            "rec_power": 0,
-            "rec_speed": 0,
-            "predicted_max_speed": 0,
-            "zdist": 0,
-            "plan_name": "",
+            "rec_power": DataValue(float),
+            "rec_speed": DataValue(float),
+            "predicted_max_speed": DataValue(float),
+            "zdist": DataValue(float),
+            "plan_name": DataValue(str),
         }
-
-        self.message = None
-        self.message_received_time = 0
-        self.message_duration = 5  # seconds
+        self.message = DataValue(str, 20)
 
     def load_message(self, message: str) -> None:
         """Store a message which is made available by self.get_message."""
-        self.message_received_time = time.time()
-        self.message = message
+        self.message.update(message)
 
     def has_message(self) -> bool:
-        """Check if a message is available for display on the overlay.
-
-        Should returns true if the message is available for display.
-
-        Returning false may mean messages have been sent, or the most
-        recent message has expired.
-        """
-        if not self.message:
-            return False
-        # Clear the message and return false if enough time has past since
-        # the message was received
-        if time.time() > self.message_received_time + self.message_duration:
-            self.message = None
-            return False
-        return True
+        """Assess whether message has expired. Will return true if expiration
+        has not bee exceeded."""
```

Comment:
Typo: `been`

Out of interest, why were these docstrings changed (`get_message` too)? I'd argue the old ones were slightly more correct/useful. This docstring doesn't consider when no messages have been sent (in which case, they technically haven't expired) and the below docstring talks about `self.message`, which is an internal variable that users of the class shouldn't have to worry about.